bundles/persistence/org.openhab.persistence.jdbc/java/org/openhab/persistence/jdbc/db/JdbcDerbyDAO.java

```diff
@@ -0,0 +1,211 @@
+		
+	}
+	
+	/**************
+	 * ITEMS DAOs *
+	 **************/
+	@Override
+	public Integer doPingDB(){
+		return Yank.queryScalar(SQL_PING_DB, Integer.class, null);
+	}
+	
+	@Override
+	public ItemsVO doCreateItemsTableIfNot(ItemsVO vo){
+		//boolean tableExists = Yank.queryScalar(SQL_IF_TABLE_EXISTS.replace("#searchTable#", vo.getItemsManageTable().toUpperCase()), String.class, null) == null; 
+		boolean tableExists = doIfTableExists(vo);
+		if(tableExists){
+			String sql = StringUtilsExt.replaceArrayMerge(
+					SQL_CREATE_ITEMS_TABLE_IF_NOT, 
+					new String[]{ "#itemsManageTable#", "#colname#", "#coltype#"}, 
+					new String[]{ vo.getItemsManageTable().toUpperCase(), vo.getColname(), vo.getColtype()}
+					);
+			logger.debug("JDBC::doCreateItemsTableIfNot tableExists={} therefore sql={}", sql);
+			Yank.execute(sql, null);
+		}else logger.debug("JDBC::doCreateItemsTableIfNot tableExists={}, did no CREATE TABLE", tableExists);
+		return vo;
+	}
+	 
+	/*************
+	 * ITEM DAOs *
+	 *************/
+	@Override
+	public void doCreateItemTable(ItemVO vo) {
+		String sql = StringUtilsExt.replaceArrayMerge(
+				SQL_CREATE_ITEM_TABLE, 
+				new String[]{ "#tableName#", "#dbType#"}, 
+				new String[]{ vo.getTableName(), vo.getDbType()}
+				);
+		Yank.execute(sql, null);
+	}
+
+	@Override
+	public void doStoreItemValue(Item item, ItemVO vo) {
+		vo = storeItemValueProvider(item, vo);
+		String sql = StringUtilsExt.replaceArrayMerge(
+				SQL_INSERT_ITEM_VALUE, 
+				new String[]{ "#tableName#", "#dbType#"}, 
+				new String[]{ vo.getTableName().toUpperCase(), vo.getDbType()}
+				);
+		Object[] params = new Object[] { vo.getValue() };
+		logger.debug("JDBC::doStoreItemValue sql={} value='{}'", sql, vo.getValue() );
+		Yank.execute(sql, params);
+	}
+	
+	@Override
+	public List<HistoricItem> doGetHistItemFilterQuery(Item item, FilterCriteria filter, int numberDecimalcount, String table, String name) {
+		String sql = histItemFilterQueryProvider(filter, numberDecimalcount, table, name);
+		logger.debug("JDBC::doGetHistItemFilterQuery sql={}", sql);
+		List<Object[]> m = Yank.queryObjectArrays(sql, null);
+
+		List<HistoricItem> items = new ArrayList<HistoricItem>();
+		for(int i=0; i<m.size(); i++){
+			items.add(new JdbcItem(item.getName(), getState(item, m.get(i)[1]), objectAsDate(m.get(i)[0]) ));
+		}
+	    return items;
+	}
+	 
+	/****************************
+	 * SQL generation Providers *
+	 ****************************/
+	/**
+	 * @param filter
+	 * @param numberDecimalcount
+	 * @param table
+	 * @return
+	 */
+	private String histItemFilterQueryProvider(FilterCriteria filter, int numberDecimalcount, String table, String simpleName) {
+		logger.debug("JDBC::getHistItemFilterQueryProvider filter = {}, numberDecimalcount = {}, table = {}, simpleName = {}", filter.toString(), numberDecimalcount, table, simpleName);
+		
+		SimpleDateFormat jdbcDateFormat = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
```

Comment:
This DateFormat object never changes. Maybe make it static final in the class scope and save a bit of performance?